src/Ecommerce.js

```diff
@@ -0,0 +1,161 @@
+        }
+    }
+    const handleNext = (max) => {
+        if(currPage !== max){
+            setCurrPage(currPage + 1);
+        }
+    }
+    const paginate = (number) => {
+        setCurrPage(number);
+    }
+
+    useEffect(() => {
+        if (products) {
+            const newProducts = [...products];
+            let filteredBrands = newProducts;
+            if(brandNames.length !== 0){
+                filteredBrands = newProducts.filter(product => {
+                    return brandNames.findIndex(brand => brand === product.brand.name) !== -1;
+                    
+                })
+            };
+            if(sort && sort === "Low to high") {
+                filteredBrands.sort((a, b) => {return a.price - b.price});
+            } else if(sort === "High to low") {
+                filteredBrands.sort((a, b) => {return b.price - a.price});
+            };
+            setTotalProducts(filteredBrands.length);
+            const filterProducts = filteredBrands.slice((currPage-1)*productsPerPage, currPage*productsPerPage);
+            setFilteredProducts(filterProducts);
+        }
+    }, [products, currPage, brandNames, sort, productsPerPage]);
+
+    const goToProduct = (id) => {
+        props.history.push('/product/' + id);
+    }
+
+    const filterByBrand = (name) => {
+    
+        const index = brandNames.findIndex(brand => brand === name);
+        let newBrands;
+        if(index === -1){
+            newBrands = [...brandNames];
+            newBrands.push(name);
+            setBrandNames(newBrands);
+        }
+        else {
+            newBrands = [...brandNames]; 
+            newBrands = newBrands.filter(brand => {
+                return brand !== name;
+            });
+            setBrandNames(newBrands);
+        };
+
+        setCurrPage(1);
+    }
+
+    const sortByPrice = (option) => {
+        setSort(option);
+        setCurrPage(1);
+    }
+
+    const changeLayout = (val) => {
+        setProductsPerPage(3*val);
+        setCurrPage(1);
+    }
+
+    return (
+        <div>
+            <main className="wrapper">
+                <div>
+                <div className="card" style={{ width: "18rem" }}>
+                    <div className="card-header">Brands</div>
+                    <ul className="list-group list-group-flush">
+                    
+                        {brands ? brands.map(brand => {
+                            return <li key={brand.id} className="list-group-item">
+                                <input type="checkbox" name="brand" onChange={() => {filterByBrand(brand.name)}}/>{brand.name}
+                            </li>;
+                        }) : null}
+
+                    </ul>
+                </div>
+                <div className="card" style={{ width: "18rem" }}>
+                    <div className="card-header">Price</div>
+                    <ul className="list-group list-group-flush">
+                        <li className="list-group-item">
```

Comment:
you should save it and the "Low to high" string as a consts, you can save only a short code for each one (like LTH or HTL)